src/gpgmm/d3d12/ResourceAllocatorD3D12.cpp

```diff
@@ -1242,18 +1242,12 @@ namespace gpgmm::d3d12 {
         resourceHeapDesc.SizeInBytes = resourceInfo.SizeInBytes;
         resourceHeapDesc.Alignment = resourceInfo.Alignment;
 
-        IHeap* resourceHeap = nullptr;
-        ReturnIfFailed(Heap::CreateHeap(
-            resourceHeapDesc, /*residencyManager*/ nullptr,
-            [&](ID3D12Pageable** ppPageableOut) -> HRESULT {
-                ComPtr<ID3D12Pageable> pageable;
-                resource.As(&pageable);
-
-                *ppPageableOut = pageable.Detach();
+        CreateResourceCallbackContext createResourceCallbackContext(resource);
```

Comment:
nit: Nothing is being created. Rename to ImportResourceCallbackContext?

Reply:
Done.